applications/nrf5340_audio/src/modules/display/log_tab.c

```diff
@@ -0,0 +1,174 @@
+
+lv_obj_t *log_list;
+int cnt;
+struct backend_context {
+	uint32_t cnt;
+	const char *exp_str[10];
+	uint32_t delay;
+	bool active;
+	struct k_timer timer;
+};
+static int cbprintf_callback(int c, void *ctx)
+{
+	char **p = ctx;
+
+	**p = c;
+	(*p)++;
+
+	return c;
+}
+
+static void panic(const struct log_backend *const backend)
+{
+	ARG_UNUSED(backend);
+}
+
+static void expire_cb(struct k_timer *timer)
+{
+	void *ctx = k_timer_user_data_get(timer);
+	struct backend_context *context = (struct backend_context *)ctx;
+
+	context->active = true;
+}
+
+static void backend_init(const struct log_backend *const backend)
+{
+	struct backend_context *context = (struct backend_context *)backend->cb->ctx;
+
+	k_timer_init(&context->timer, expire_cb, NULL);
+	k_timer_user_data_set(&context->timer, (void *)context);
+	k_timer_start(&context->timer, K_MSEC(context->delay), K_NO_WAIT);
+}
+
+static int backend_is_ready(const struct log_backend *const backend)
+{
+	struct backend_context *context = (struct backend_context *)backend->cb->ctx;
+
+	return context->active ? 0 : -EBUSY;
+}
+
+static void add_log_msg_to_log_list(lv_obj_t *log_obj, char *log_msg, lv_color_t log_msg_color)
+{
+	log_obj = lv_list_add_text(log_obj, log_msg);
+	lv_obj_set_style_bg_color(log_obj, lv_palette_darken(LV_PALETTE_GREY, 3), 0);
+	lv_obj_set_style_text_color(log_obj, log_msg_color, 0);
+	lv_label_set_long_mode(log_obj, LV_LABEL_LONG_WRAP);
+	lv_obj_scroll_to_view(log_obj, LV_ANIM_OFF);
+}
+
+static void backend_process(const struct log_backend *const backend, union log_msg_generic *msg)
+{
+	log_timestamp_t msg_timestamp;
+	int msg_timestamp_in_us;
+	char str[200];
+	char timestamp_n_str[220];
+	char *pstr = str;
+
+	lv_obj_t *oldest_label;
+
+	struct backend_context *context = (struct backend_context *)backend->cb->ctx;
+	size_t len;
+	uint8_t *p = log_msg_get_package(&msg->log, &len);
+
+	int slen = cbpprintf(cbprintf_callback, &pstr, p);
+	str[slen] = '\0';
+
+	if (IS_ENABLED(CONFIG_LOG_TIMESTAMP)) {
+		msg_timestamp = log_msg_get_timestamp(&msg->log);
+		msg_timestamp_in_us = log_output_timestamp_to_us(msg_timestamp);
+		snprintf(timestamp_n_str, sizeof(timestamp_n_str), "[%d] %s", msg_timestamp_in_us,
+			 str);
+		strcpy(str, timestamp_n_str);
+	}
+
+	k_mutex_lock(&nrf5340_display_mtx, K_FOREVER);
+
+	switch (log_msg_get_level(&msg->log)) {
+	case LOG_LEVEL_ERR:
+		if (CONFIG_DISPLAY_LOG_LEVEL >= 1) {
+			cnt++;
+			add_log_msg_to_log_list(log_list, str,
+						lv_palette_lighten(LV_PALETTE_RED, 1));
+		}
+		break;
+	case LOG_LEVEL_WRN:
+		if (CONFIG_DISPLAY_LOG_LEVEL >= 2) {
+			cnt++;
+			add_log_msg_to_log_list(log_list, str,
+						lv_palette_lighten(LV_PALETTE_YELLOW, 1));
+		}
+		break;
+
+	case LOG_LEVEL_INF:
+		if (CONFIG_DISPLAY_LOG_LEVEL >= 3) {
+			cnt++;
+			add_log_msg_to_log_list(log_list, str,
+						lv_palette_lighten(LV_PALETTE_YELLOW, 5));
+		}
+		break;
+	case LOG_LEVEL_DBG:
+		if (CONFIG_DISPLAY_LOG_LEVEL >= 4) {
+			cnt++;
```

Comment:
Change to something more descriptive. E.g.:
```suggestion
			num_log_messages++;
```

Reply:
This has been updated to `log_list_length`